api/src/logger/logger.service.ts

```diff
@@ -1,11 +1,87 @@
 /*
- * Copyright © 2024 Hexastack. All rights reserved.
+ * Copyright © 2025 Hexastack. All rights reserved.
  *
  * Licensed under the GNU Affero General Public License v3.0 (AGPLv3) with the following additional terms:
  * 1. The name "Hexabot" is a trademark of Hexastack. You may not use this name in derivative works without express written permission.
  * 2. All derivative works must include clear attribution to the original creator and software, Hexastack and Hexabot, in a prominent location (e.g., in the software's "About" section, documentation, and README file).
  */
 
-import { ConsoleLogger } from '@nestjs/common';
+import {
+  ConsoleLogger,
+  Inject,
+  Injectable,
+  LogLevel,
+  Scope,
+} from '@nestjs/common';
+import { INQUIRER } from '@nestjs/core';
 
-export class LoggerService extends ConsoleLogger {}
+@Injectable({ scope: Scope.TRANSIENT })
+export class LoggerService extends ConsoleLogger {
+  private logLevels: LogLevel[] = [];
+
+  constructor(@Inject(INQUIRER) private parentClass: object) {
+    super(parentClass.constructor.name);
+    this.initLogLevels();
```

Comment:
Can you please check the 2nd param which is options, I think in there you can control the log level at the parent class level. Can you check?